lib/note-content-editor.jsx

```diff
@@ -63,6 +120,54 @@ export default class NoteContentEditor extends React.Component {
 		invoke( this, 'editor.focus' );
 	}
 
+	onTab = ( e ) => {
+		// prevent moving focus to next input
+		e.preventDefault()
+
+		const editorState = this.state.editorState;
+		const selection = editorState.getSelection();
+
+		if ( ! selection.isCollapsed() ) {
+			return
+		}
+
+		const content = editorState.getCurrentContent();
+		const selectionStart = selection.getStartOffset();
+		const block = content.getBlockForKey( selection.getFocusKey() );
+		const line = block.getText();
+
+		const atStart = line.trim() === '-' || line.trim() === '*';
+
+		if ( ! e.shiftKey ) {
+			// inserting a tab character
+
+			const offset = atStart ? 0 : selectionStart
+
+			const newEditorState = EditorStateModifier.insertText(
+				editorState,
+				selection.merge( { anchorOffset: offset, focusOffset: offset } ),
+				'\t'
+			);
+
+			this.handleEditorStateChange( newEditorState );
+		} else {
+			// outdenting
+
+			const rangeStart = atStart ? 0 : selectionStart - 1;
+			const rangeEnd = atStart ? 1 : selectionStart;
+			const prevChar = block.getText().slice( rangeStart, rangeEnd );
+
+			if ( prevChar === '\t' ) {
+				const newEditorState = EditorStateModifier.removeRange(
+					editorState,
+					selection.merge( { anchorOffset: rangeStart, focusOffset: rangeEnd } )
+				);
+
+				this.handleEditorStateChange( newEditorState );
+			}
```

Comment:
I probably commented on this in the other PR, so again, sorry for the dupes.

can we split these into separate `indentRange` and `outdentRange` functions? they don't have to hold those names, but it would be nice to have them in separate functions. further, after making sure that other modifier keys aren't pressed (<kbd>ctrl</kbd>, <kbd>meta</kbd>, <kbd>alt</kbd>), we can simplify this function here to

```js
return this.handleEditorStateChange(
	e.shiftKey
		? outdentRange( editorState )
		: indentRange( editorState )
);
```

the general principle here is twofold:
 - keep functions as short and single-minded as possible
 - keep event handlers even shorter so that they are primarily dispatchers to other functions instead of functions themselves which do things. (we should be able to easily indent or outdent a range without having to call the `onTab` handler)